File: examples/upload-download-files-oauth2-php-api/src/Controller/ProjectsController.php
<?php

namespace App\Controller;

use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
use Symfony\Component\HttpFoundation\Request;
use Symfony\Component\HttpFoundation\Response;
use Symfony\Component\HttpFoundation\Session\SessionInterface;
use Symfony\Component\Routing\Annotation\Route;

class ProjectsController extends OAuthAuthenticatedController
{
    #[Route('/projects', name: 'list_projects')]
    public function index(): Response
    {
        if (!$this->checkToken()) {
            return $this->redirect($this->generateUrl('oauth_home'));
        }

        $projects = $this->client->request('GET', $this->getParameter('lokalise.api.endpoint') . 'projects', [
                'query' => [
                    'limit' => 50
                ]
            ])?->toArray() ?? [];

        $projects = array_map(fn($data) => ['id' => $data['project_id'], 'name' => $data['name']], $projects['projects']);

        return $this->render('projects/index.html.twig', [
            'projects' => $projects,
        ]);
    }

    #[Route('/select_project/{projectId}', name: 'oauth_select_project')]
    public function select(SessionInterface $session, string $projectId): Response
    {
        if ($this->checkToken()) {
            $session->set('lokalise.oauth.project_id', $projectId);
        }

        return $this->redirect($this->generateUrl('oauth_home'));
    }
}

File: examples/upload-download-files-oauth2-rails-api/app/controllers/projects_controller.rb
# frozen_string_literal: true

require 'ruby_lokalise_api'

class ProjectsController < ApplicationController
  before_action :check_token

  def index
    client = RubyLokaliseApi.oauth2_client session[:lokalise_token]
    @projects = client.projects(limit: 5000).collection.map { |p| { id: p.project_id, name: p.name } }
  end

  def choose
    session[:lokalise_project_id] = params[:project_id]
    redirect_to new_oauth2_flow_path
  end
end
